src/wasm/wat-parser.cpp

```diff
@@ -32,11 +32,11 @@
 // definitions. This phase establishes the indices and names of each module
 // element so that subsequent phases can look them up.
 //
-// The second phase, not yet implemented, parses type definitions to construct
-// the types used in the module. This has to be its own phase because we have no
-// way to refer to a type before it has been built along with all the other
-// types, unlike for other module elements that can be referred to by name
-// before their definitions have been parsed.
+// The second phase parses type definitions to construct the types used in the
+// module. This has to be its own phase because we have no way to refer to a
+// type before it has been built along with all the other types, unlike for
+// other module elements that can be referred to by name before their
+// definitions have been parsed.
```

Comment:
Isn't necessarily related to this PR, but I think you planned to make some phases, this or the function code parsing, parallel. Is this not the phase you wanted to parallelize?

Reply:
I was only planning on parallelizing phase 5 since that's the one that constructs function bodies, but I guess in principle we could parallelize phases 2 and 4 as well.